host_core/lib/host_core/wasmcloud/runtime.ex

```diff
@@ -0,0 +1,97 @@
+defmodule HostCore.WasmCloud.Runtime do
+  @moduledoc """
+  This module provides encapsulated access to the native NIF function calls exposed through the wasmcloud/native.ex interface
+  This runtime module supports the Runtime.Server module and should _not_ be used directly by any code other than that module. This
+  would be a private module, but Elixir doesn't allow selective access
+  """
+  @type t :: %__MODULE__{
+          resource: binary(),
+          reference: reference()
```

Comment:
So, an actor having a resource (the bytes of the actor) and a reference (reference to actor) makes sense. What's the resource of a runtime?

Reply:
the resource is actually the resource arc that comes from the NIF:
```
  let rt = WcRuntime::from_host_handler(host_handler)
        .context("failed to construct runtime")
        .map_err(|e| Error::Term(Box::new(e.to_string())))?;

    let resource = ResourceArc::new(RuntimeResource { inner: rt });
    Ok(resource)
```

The use of the binary here is an esoteric incantation required to keep the various compilers from inlining and removing data